pkg/operator/ceph/object/realm/controller.go

```
@@ -162,10 +170,24 @@ func (r *ReconcileObjectRealm) reconcile(request reconcile.Request) (reconcile.R
 	// Start object reconciliation, updating status for this
 	updateStatus(r.client, request.NamespacedName, k8sutil.ReconcilingStatus)
 
-	// CREATE/UPDATE CEPH REALM
-	reconcileResponse, err = r.createCephRealm(cephObjectRealm)
-	if err != nil {
-		return r.setFailedStatus(request.NamespacedName, "failed to create ceph realm", err)
+	// Create/Pull Ceph Realm
+	if cephObjectRealm.Spec.IsPullRealm() {
+		logger.Debug("pull section in spec found")
+		reconcileResponse, err = r.pullCephRealm(cephObjectRealm)
+		if err != nil {
+			return r.setFailedStatus(request.NamespacedName, "failed to pull ceph realm", err)
```

@travisn do you think I should requeue the realm pull here or just let it fail?